Wikipedia/Code/WMFArticleFetcher.m

```diff
@@ -41,6 +41,67 @@ @interface WMFArticleBaseFetcher ()
 
 @implementation WMFArticleBaseFetcher
 
++ (NSOperationQueue *)articleSaveQueue {
+    static dispatch_once_t onceToken;
+    static NSOperationQueue *articleSaveQueue;
+    dispatch_once(&onceToken, ^{
+        articleSaveQueue = [NSOperationQueue new];
+        articleSaveQueue.maxConcurrentOperationCount = 1;
+    });
+    return articleSaveQueue;
+}
+
++ (NSMutableDictionary<NSString *, NSOperation *> *)articleSaveOperations {
+    static dispatch_once_t onceToken;
+    static NSMutableDictionary<NSString *, NSOperation *> *articleSaveOperations;
+    dispatch_once(&onceToken, ^{
+        articleSaveOperations = [NSMutableDictionary new];
+    });
+    return articleSaveOperations;
+}
+
++ (void)asynchronouslySaveArticle:(MWKArticle *)article {
+    NSOperationQueue *queue = [self articleSaveQueue];
+    NSMutableDictionary *operations = [self articleSaveOperations];
+    @synchronized (queue) {
+        NSString *key = article.url.wmf_databaseKey;
+        if (!key) {
+            return;
+        }
+        
+        NSOperation *op = operations[key];
+        if (op) {
+            [op cancel];
```

Comment:
my thought was if the later save had an updated version of the article, we should have that one go through